src/lists/list-item.jsx

```diff
@@ -224,8 +224,7 @@ let ListItem = React.createClass({
     React.Children.forEach(this.props.children, (child) => {
       if (child === null) return;
 
-      //consider all element to be nested except literals.
-      if (React.isValidElement(child)) {
+      if (React.isValidElement(child) && child.type.displayName === 'ListItem') {
```

Comment:
as far I remember, I check with the docs, nested was working. My use case I to use the list item with textfields, selectfields and customs component inside.